windows/sound.cpp

```diff
@@ -80,15 +87,26 @@ QString CSound::LoadAndInitializeDriver ( QString strDriverName,
     // check if device is capable
     if ( strStat.isEmpty() )
     {
-// TODO: In order to fix https://github.com/jamulussoftware/jamulus/issues/796 we reset the channel mapping on every property change. This is not ideal. 
-	    
-// the device has changed, per definition we reset the channel
-// mapping to the defaults (first two available channels)
-ResetChannelMapping();
-
-// store ID of selected driver if initialization was successful
-strCurDevName = cDriverNames[iDriverIdx];
-
+        // Reset channel mapping if the number of channels has changed or the sound card name has changed
+        if ( ( lNumInChanPrev != lNumInChan ) ||
+             ( lNumOutChanPrev != lNumOutChan ) ||
+             ( strCurDevName.compare ( strDriverNames[iDriverIdx] ) != 0 ) )
+        {
+            // TODO: In order to fix https://github.com/jamulussoftware/jamulus/issues/796
+            // we reset the channel mapping if the number of channels has changed or the device
+            // name has changed. This is not ideal.
+
+            // the device has changed, per definition we reset the channel

```

Comment:
I'd add a clarification here: this code runs after a change in the ASIO driver -- not when changing the ASIO input selection.  Hence, if I switch from "ASIO4ALL with four inputs" to "Echo AudioFire4 with four inputs", it will run -- I'll then have to switch to inputs 3&4 on the AudioFire, if that's what I want, as `ResetChannelMapping();` will revert from any previous settings I might have had.

(Maybe we need an array of settings per soundcard to get this working pleasantly...)

Reply:
> (Maybe we need an array of settings per soundcard to get this working pleasantly...)

Maybe in future